lib/jekyll/collection.rb

```diff
@@ -57,18 +57,9 @@ def read
         full_path = collection_dir(file_path)
         next if File.directory?(full_path)
         if Utils.has_yaml_header? full_path
-          doc = Jekyll::Document.new(full_path, { :site => site, :collection => self })
-          doc.read
-          if site.publisher.publish?(doc) || !write?
-            docs << doc
-          else
-            Jekyll.logger.debug "Skipped From Publishing:", doc.relative_path
-          end
+          read_docs(full_path)
```

Comment:
In this case, you're just reading one document. Thus, this should be singular: `read_document(full_path)`.